docs/transmission.md

```diff
@@ -0,0 +1,89 @@
+# Transmission model
+
+## Overview
+
+The transmission model controls person-to-person transmission between an
+infectious agent and other susceptible agents. Agents become infectious immediately
+upon successful infection attempt, though they may have a transmission probability of
+0 for some time (mimicking an exposed compartment). Once infectious, agents have a pre-assigned
+number of infection attempts. Infection attempt times are drawn from the generation interval.
+Once there are no more infection attempts remaining for the agent, we argue that they do not
+transition to a recovered compartment but rather back to a susceptible compartment but
+with high immunity.
+
+## Workflow
+
+The transmission model is centered around a key workflow:
+1. Watch for an agent becoming infectious (S --> I transition).
+2. For the infectious agent, draw a number of infection attempts. This can be from the
+$R_i$ distribution or the contact distribution.
+3. Draw an infection time for the agent. This consists of taking an ordered draw from the generation
+interval. The math for this is abstracted away under a generic `get_next_infection_time` method.
+4. Draw a susceptible contact from the agent's contact group. Contacts could be drawn uniformly
+from the contact group or with weight based on person characteristics. The actual
+drawing of a contact can be abstracted into a `get_contact` method. Evaluate whether
+the infection attempt is successful. Success is based on:
+    - a potential probability of transmission in the case where we are drawing the
+    number of contacts from a contact distribution,
+    - whether any interventions are present that may reduce transmission,
+    - the contact's immunity level which is updated when the contact is drawn.
+
+    If the infection attempt is successful, the infectious contact is labeled as infectious.
+5. Repeat steps #3 and #4, decreasing the number of infection attempts left by one each time
+regardless of whether the attempt is successful and ending when the agent has no zero
+infection attempts remaining.
+6. Re-label the agent as susceptible. The immunity manager will set their latent immunity level
```

Comment:
This isn't entirely clear to me. Why not recovered? If we are not infecting anyone else why wouldn't we move to the next compartment?

Reply:
The thinking here was that we would only want the transmission manager to have infectious or susceptible compartments, with susceptible individuals having a susceptibility determined by recent infections, innate immunity, etc, that is altered through the immunity manager. We therefore don't necessarily need a recovered compartment and it simplifies including those post-infectious individuals to be in the contact network of future infectors. The person, now susceptible, wouldn't infect anyone else